source/gui/__init__.py

```diff
@@ -484,10 +484,22 @@ def __init__(self, frame: MainFrame):
 			self.Bind(wx.EVT_MENU, lambda evt: os.startfile("http://www.nvda-project.org/"), item)
 			# Translators: The label for the menu item to view NVDA License document.
 			item = menu_help.Append(wx.ID_ANY, _("L&icense"))
-			self.Bind(wx.EVT_MENU, lambda evt: os.startfile(getDocFilePath("copying.txt", False)), item)
+			# Using operation="edit" to os.startfile below, to work around a Windows 11 related
+			# bug in Python 3.7's os.startfile. (#14725) Test reverting to original behavior after Python upgrade.

```

Comment:
Can this comment and the logic responsible for opening text files be moved to a separate function in `systemUtils`? This ensures that both code and  the accompanying comment is not duplicated, and provides add-on developers with a tested way for opening text files under all versions of Windows.

Reply:
It certainly can be. However I didn't do that, though I considered it, for the following reasons:

1. This is hopefully a temporary condition that will be resolved by upgrading Python later this year. If so, the need for such a specific function will go away, and we will revert this. If instead we create a dedicated function for text files only, we'll likely be stuck with it for a long time, even if internally we change its behavior to be identical to `os.startfile`'s intended use once this bug is resolved.
2. If add-ons start using such a function assuming it is "the preferred way", which opens for editing as a workaround instead of opening for opening, it increases the chance of unexpected behavior. Better for add-on devs to call the edit operation directly, so they know their doing something slightly unusual, even if likely harmless, and can stop doing it when we revert this. It draws attention to the fact that this **is a workaround**.
3. Opening all other file types will still be done with `os.startfile`. I question the value in fragmenting the file opening modalities.

All that said, I will follow whatever @seanbudd  or @michaelDCurran  think is best practice.

Reply:
> If instead we create a dedicated function for text files only, we'll likely be stuck with it for a long time, even if internally we change its behavior to be identical to os.startfile's intended use once this bug is resolved.

I don't think we would be stuck with it, I would encourage using an underscored function for this, so we can remove it when `os.startfile` is fixed. We would also not be upgrading python outside of an API changing release, so being stuck with a helper function is a non-issue.
Using a private function also prevents the issue raised with add-on misuse.

I have been unable to find a bug reported against Python or Windows related to this. 
Without an external bug to track, I don't think we can assume this will be fixed in a future release.
A helper function makes tracking and using this workaround easier in the future.

Reply:
@seanbudd I was mainly thinking in terms of the public function @lukaszgo1 was asking for, so that add-on devs could use it. A private function makes more sense to me than a public one does.